File: Power.java
import java.nio.file.Path;

import com.onthegomap.planetiler.FeatureCollector;
import com.onthegomap.planetiler.Planetiler;
import com.onthegomap.planetiler.Profile;
import com.onthegomap.planetiler.config.Arguments;
import com.onthegomap.planetiler.reader.SourceFeature;

public class Power implements Profile {

  @Override
  public void processFeature(SourceFeature sourceFeature, FeatureCollector features) {
    if (sourceFeature.canBeLine() && sourceFeature.hasTag("power", "line")) {
      features.line("power")
          .inheritAttrFromSource("voltage")
          .setMinZoom(7);
    }
  }

  public static void main(String[] args) throws Exception {
    Planetiler.create(Arguments.fromArgs(args))
        .addOsmSource("osm", Path.of("ri.osm.pbf"), "geofabrik:rhode-island")
        .overwriteOutput(Path.of("power.pmtiles"))
        .setProfile(new Power())
        .run();
  }

}

File: power.lua
function planetiler.process_feature(source, features)
  if source:has_tag('power', 'line') then
    features:line('power')
        :inherit_attr_from_source('voltage')
        :set_min_zoom(7)
  end
end

planetiler.output.path = 'power.pmtiles'
planetiler.examples = 'power.spec.yml'
planetiler.add_source('osm', {
  type = 'osm',
  url = 'geofabrik:rhode-island'
})
